### PR TITLE
RAP-4092: added doPost and some cleanup

### PR DESCRIPTION
**Pull request by jonathan-major**

### CODE REVIEW

File: Libs/RaptureWebServlet/src/main/java/rapture/server/web/servlet/UserBoundedScriptPageServlet.java [hunk 1]
```diff
@@ -213,4 +301,45 @@ protected String getReflexScript(String uri) {
...
+     * @return      Map of HTTP header 
+     */
+    private Map<String, Object> getRequestVariables(HttpServletRequest req) {
+        Map<String, Object> serverMap = new HashMap<String, Object>();
+        serverMap.put("ContentType", req.getContentType());
+        serverMap.put("ContextPath", req.getContextPath());
+        serverMap.put("Method", req.getMethod());
+        serverMap.put("PathInfo", req.getPathInfo());
+        serverMap.put("RemoteAddr", req.getRemoteAddr());
+        serverMap.put("", req.getLocalAddr());
```
Comment (dukenguyen) [inline Libs/RaptureWebServlet/src/main/java/rapture/server/web/servlet/UserBoundedScriptPageServlet.java]: typo?

File: Libs/RaptureWebServlet/src/main/java/rapture/server/web/servlet/UserBoundedScriptPageServlet.java [hunk 2]
```diff
@@ -193,6 +246,35 @@ protected void runScriptWithUser(String script, Map<String, Object> parameterMap
...
+    /**
+     * Helper method to extract the URI from HttpServletRequest
+     * <p>
+     * @param  request  The actual script/endpoint to execute  
+     * @return          URI of script as a String e.g. script://<repo>/endpoint 
+     */
+    private String getScriptUrlFromHttpRequest(HttpServletRequest request){
+        String scriptURL = "";
+        
+        String path = request.getPathInfo();
```
Comment (dukenguyen) [inline Libs/RaptureWebServlet/src/main/java/rapture/server/web/servlet/UserBoundedScriptPageServlet.java]: getPathInfo from HttpServletRequest can return null.  From the javadoc:

"This method returns null if there was no extra path information."

Might be good to check if its null before split()ing.

File: Libs/RaptureWebServlet/src/main/java/rapture/server/web/servlet/UserBoundedScriptPageServlet.java [hunk 3]
```diff
@@ -193,6 +246,35 @@ protected void runScriptWithUser(String script, Map<String, Object> parameterMap
...
     }
 
+    /**
+     * Helper method to extract the URI from HttpServletRequest
+     * <p>
+     * @param  request  The actual script/endpoint to execute  
+     * @return          URI of script as a String e.g. script://<repo>/endpoint 
+     */
+    private String getScriptUrlFromHttpRequest(HttpServletRequest request){
+        String scriptURL = "";
```
Comment (dukenguyen) [inline Libs/RaptureWebServlet/src/main/java/rapture/server/web/servlet/UserBoundedScriptPageServlet.java]: declare this when its first used on line 265.  no need to initialize to empty string here.

File: Libs/RaptureWebServlet/src/main/java/rapture/server/web/servlet/UserBoundedScriptPageServlet.java (hunk 2 above)
Comment (jonathan-major) [inline Libs/RaptureWebServlet/src/main/java/rapture/server/web/servlet/UserBoundedScriptPageServlet.java]: updated method to check for null and added checks in calling methods.

**Pull request merged**